config.go

```diff
@@ -117,9 +119,63 @@ func makeDockerV2S2Image(oimage *ociv1.Image) (docker.Image, error) {
 	return image, nil
 }
 
+// makeDockerV2S1Image builds the best docker image structure we can from the
+// contents of the V2S1 image structure.
+func makeDockerV2S1Image(manifest docker.V2S1Manifest) (docker.V2Image, error) {
+	// Treat the most recent (first) item in the history as a description of the image.
+	if len(manifest.History) == 0 {
+		return docker.V2Image{}, fmt.Errorf("error parsing image configuration from manifest")
+	}
+	dimage := docker.V2Image{}
+	err := json.Unmarshal([]byte(manifest.History[0].V1Compatibility), &dimage)
+	if err != nil {
+		return docker.V2Image{}, err
+	}
+	if dimage.DockerVersion == "" {
+		return docker.V2Image{}, fmt.Errorf("error parsing image configuration from history")
+	}
+	// Build a filesystem history.
+	rootFS := &docker.V2S2RootFS{
+		Type:    docker.TypeLayers,
+		DiffIDs: []digest.Digest{},
+	}
+	history := []docker.V2S2History{}
+	for i := range manifest.FSLayers {
+		// The DiffID list is intended to contain the sums of _uncompressed_ blobs, so if the blob is
```

Comment:
Blobs can be either data (such as a JSON-encoded config blob) or layers, and layers can be uncompressed or compressed.  Compression is usually done gzip or bzip2.  I expect bzip2 is rare (I've literally never seen it used) because the Go compress/bzip2 implementation can decompress but not compress.  Most layers in the wild are compressed with gzip compression.

For Docker v2 schema 2, they're tagged with one of the MIME types listed in https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md.  In OCI v1, they're tagged with one of the MIME types listed at https://github.com/opencontainers/image-spec/blob/master/media-types.md.  For Docker v2 schema 1, I think we don't have that information, but we can look at the initial segment of the blob and make a guess.  Problem is that we don't want to take the time to decompress a possibly-very-large blob to calculate the correct value, so I'm starting to lean toward @mtrmac's suggestion that it's better to leave the DiffID list empty to avoid creating errors if we ever try to use this data.